src/kakarot/instructions/system_operations.cairo

```diff
@@ -94,4 +95,44 @@ namespace SystemOperations {
             ctx, new_return_data_len=size.low, new_return_data=new_return_data
         );
     }
+
+    // @notice REVERT operation.
+    // @dev
+    // @custom:since Byzantium
+    // @custom:group System Operations
+    // @custom:gas 0 + dynamic gas
+    // @custom:stack_consumed_elements 0
+    // @custom:stack_produced_elements 0
+    // @return The pointer to the updated execution context.
+    func exec_revert{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(ctx: model.ExecutionContext*) {
+        alloc_locals;
+
+        // Get stack and memory from context
+        let stack = ctx.stack;
+        let memory = ctx.memory;
+
+        // Stack input:
+        // 0 - size: byte size to copy
+        // 1 - offset: byte offset in the memory in bytes
+        let (stack, popped) = Stack.pop_n(self=stack, n=2);
+        // TODO: implement loading of the revert reason based on size value,
```

Comment:
why isn't possible to use `size`?

Reply:
In this implementation as we're using StarkNet's revert so I thought on limiting to loading from the offset as the short string can't be longer than 32 bytes. But let me know if there could be a smarter work around, not 100% sure about this

Reply:
fair